feat: Add reference information and how-to guide for Glance (#208)
citynetwork/docs: docs/reference/images/index.md

---

> description: Images provide an Operating System for your server.

Reviewer: ```suggestion
description: Images provide an operating system for your server.
```

---

# Images

Public [Glance images](../../howto/openstack/glance/index.md) in {{brand}} contain regularly updated minimal versions of server operating systems.

> These images also contain the [cloud-init](https://cloudinit.readthedocs.io) package applicable to the operating system, to support the injection of SSH public keys and other and user data.

Reviewer: ```suggestion
These images also contain the [cloud-init](https://cloudinit.readthedocs.io) package applicable to the operating system, to support the injection of SSH public keys and other user data.
```

## Naming conventions

### Image names

Image names in {{brand}} follow a convention, which can be summarized as `${NAME} ${VERSION_ID} ${CODENAME} ${ARCH}`:

* `NAME`: Operating System name, such as `Ubuntu`, `Debian`, `Rocky`, etc.
* `VERSION_ID`: Operating System version, as in `22.04`, `11`, `9`, etc.

> * `CODENAME`: Operating System codename, if present, for example: `Jammy Jellyfish`, `Bullseye`, etc.

Author: Yeah nah. :) We'll keep this to images we actually make available.

> * `CODENAME`: Operating System codename, if present, for example: `Jammy Jellyfish`, `Bullseye`, etc.

Reviewer: Uhm, sorry, I meant to say "Blue Onyx" (Rocky 9, https://git.rockylinux.org/original/rpms/rocky-release/-/blob/r8/code).

> * `CODENAME`: Operating System codename, if present, for example: `Jammy Jellyfish`, `Bullseye`, etc.

Author: @noonedeadpunk Any thoughts on why we *don't* include the code name in our Rocky images?

> * `CODENAME`: Operating System codename, if present, for example: `Jammy Jellyfish`, `Bullseye`, etc.

Reviewer 2: To be frank I didn't know it has a codename - thought it's in line with CentOS that didn't, and never saw rocky folks referring it by codename, so just inertia of thinking is the reason :(

> * `CODENAME`: Operating System codename, if present, for example: `Jammy Jellyfish`, `Bullseye`, etc.

Reviewer: I also heard of no one, ever, mentioning Rocky by a codename. But then I thought there must be one, so I went hunting on Wikipedia (https://en.wikipedia.org/wiki/Rocky_Linux).

> * `CODENAME`: Operating System codename, if present, for example: `Jammy Jellyfish`, `Bullseye`, etc.

Author: @noonedeadpunk Okay, I guess it's fine then that we keep the "if present" bit but not mention any codename for Rocky, since our images names don't include it either.

> * `CODENAME`: Operating System codename, if present, for example: `Jammy Jellyfish`, `Bullseye`, etc.

Reviewer 2: We can add codenames to rocky as well, if we think it's correct and worth doing

>    Modified code names may reflect that some images serve special purposes, such as vGPU support.

Reviewer: ```suggestion
   Modified codenames may reflect that some images serve special purposes, such as vGPU support.
```

   See below for details.

> * `ARCH`: Platform architechture, for which Operating System was built, for example: `x86_64`, `aarch64`, etc.

Reviewer: ```suggestion
* `ARCH`: Platform architecture for which the operating system was built, for example, `x86_64`, `aarch64`, etc.
```

### Special codenames

* `NVGRID`: Images with `NVGRID` in the codename include pre-installed Nvidia GRID drivers and license files.
  These images allow you to use [CUDA](https://en.wikipedia.org/wiki/CUDA), and enable other computational features of vGPUs.
  You should use these images together with [flavors](../flavors/index.md) using the `g` [compute tier](../flavors/index.md#Compute_tiers).

## Tags and properties

Each public image is assigned a specific set of tags and properties.
You can use these tags to [filter and list](../../howto/openstack/glance/filter.md) images based on certain conditions.

### Tags

All public images available in {{brand}} support the following image tags:

* `os:${NAME}`: a short identifier for the operating system, such as `os:ubuntu`, `os:debian`, `os:rocky`, etc.
* `os_version:${VERSION_ID}`: the operating system version, such as `os_version:22.04`, `os_version:11`, `os_version:9`, etc.

### Properties

All public images available in {{brand}} support the following image properties:

* `architecture=${ARCH}`: Platform architecture, such as `architecture=x86_64`
* `os_distro=${NAME}`: distribution name, such as `os_distro=ubuntu`

> * `os_version=${VERSION_ID}`: operating system, such as `os_version=22.04`

Reviewer: ```suggestion
* `os_version=${VERSION_ID}`: operating system version, such as `os_version=22.04`
```
